helix-lsp/src/client.rs

```diff
@@ -1114,6 +1115,31 @@ impl Client {
         Some(self.call::<lsp::request::InlayHintRequest>(params))
     }
 
+    pub fn text_document_color_swatches(
+        &self,
+        text_document: lsp::TextDocumentIdentifier,
+        work_done_token: Option<lsp::ProgressToken>,
+    ) -> Option<impl Future<Output = Result<Value>>> {
+        let capabilities = self.capabilities.get().unwrap();
```

Comment:
Just making sure, but is `.unwrap()` safe here? Should we maybe return a `None` if there is nothing to unwrap?

Reply:
Other methods also use `.unwrap()`. See `text_document_range_inlay_hints`, `resolve_code_actions` and others

There may be some specific reasons why we need to do it this way but I don't know, since it doesn't seem to be documented anywhere

I think it's fine, since all of these methods use `unwrap` this one can use it too

Reply:
My only concern is users getting an error that should actually be an information level log (maybe you just dont have an LSP that supports this capability for rust for example, not an error in particular)

Reply:
> My only concern is users getting an error that should actually be an information level log (maybe you just dont have an LSP that supports this capability for rust for example, not an error in particular)

Oh, I see! The thing is none of the other methods will log anything so I don't think this one should either

Reply:
That's fine. Just making sure we dont display issues when there are none.

Sorry for the insistent nitpicking, trying to remember all the intricacies😅